x-pack/plugins/profiling/public/components/flame_graphs_view/index.tsx

```diff
@@ -261,7 +262,27 @@ export function FlameGraphsView({ children }: { children: React.ReactElement })
               </EuiFlexGroup>
             </EuiPanel>
           </EuiFlexItem>
-        ) : null}
+        ) : (
+          <EuiFlexItem grow={false}>
+            <EuiPanel hasShadow={false} color="subdued">
+              <PrimaryProfilingSearchBar />
+              <EuiHorizontalRule />
+              <EuiFlexGroup direction="row">
+                <EuiFlexItem grow style={{ alignItems: 'flex-end' }}>
+                  <EuiSwitch
+                    checked={showInformationWindow}
+                    onChange={() => {
+                      setShowInformationWindow((prev) => !prev);
+                    }}
+                    label={i18n.translate('xpack.profiling.flameGraph.showInformationWindow', {
+                      defaultMessage: 'Show information window',
+                    })}
+                  />
```

Comment:
This label is used twice, I'm actually surprised this passes CI. Maybe because the defaultMessage is the same. In any case, this probably makes more sense as a separate component. (Probably also good to extract the differential controls out into a separate component, that way you can more easily collapse the layout here, so you don't have to define e.g. `EuiPanel` twice. But not a huge issue, up to you.

Reply:
Thanks, I decided to create two components to flatten the layout and reduce duplication: one for the information window toggle and another for the search panel above both flamegraphs. I simplified the differential-specific code without needing to pass more state than necessary across components.

Another refactoring pass might be useful later, but I think this is good enough for now.